### ISSUE TITLE
vscode: manage extensions/settings without installing package

### ISSUE THREAD

**Issue by kubukoz** [nix-community/home-manager#3375]
### Description

I'm using the vscode module of HM to install extensions and configure settings, but I don't want to manage the program's installation itself with Nix (I want it to be self-updating, so I got it from brew). Could this be allowed by the module?

For now, I use this as a workaround:

```nix
{
  programs.vscode.package = pkgs.runCommand "dummy" { } "mkdir $out" // { pname = pkgs.vscode.pname; };
}
```

**Comment by eclairevoyant** [comment]
This isn't specific to the vscode module. If you want to change how a program installs, you'll inevitably be using something like you showed or overlays (see https://github.com/nix-community/home-manager/issues/1375#issuecomment-653747701). To me this works as intended

**Comment by ncfavier** [comment]
`pkgs.emptyDirectory` should work, but yeah, we should probably allow `package` options to be `null` in general. That would have to be coordinated with nixpkgs' `mkPackageOption` though.

**Comment by ncfavier** [comment]
https://github.com/NixOS/nixpkgs/pull/232808

**Comment by ncfavier** [comment]
I just realised there's a problem with this: the vscode module relies on `cfg.package` to determine the "flavour" (code/insiders/codium) and version of the package, which influences the location of the configuration and extension folders and the way the extensions are managed (a change in 1.74.0 required special handling). So simply allowing the option to be `null` wouldn't work.

You worked around this by adding a `pname` to your dummy package, and we could possibly add options `vscodeName` and `vscodeVersion` to allow specifying this independently from `package`, but I'm not sure this is worth it.

**Comment by kubukoz** [comment]
fair enough, this isn't a blocker for me - the current state happens to work. It's a "nice to have" :)

**Comment by eclairevoyant** [comment]
> I just realised there's a problem with this: the vscode module relies on `cfg.package` to determine the "flavour" (code/insiders/codium) and version of the package

Which is a design worth fixing as it should be handled by e.g. an option to specify the flavour

**Comment by OliverJAsh** [comment]
I'm using the workaround mentioned above but I get this error. @kubukoz Did you find a solution for this?:

```
Activating installPackages
Activating onFilesChange
/nix/store/zhrjg6wxrxmdlpn6iapzpp2z2vylpvw5-home-manager.sh: line 125: /nix/store/jply1gcrcc4q48055849ai4w4harw28q-dummy/bin/vscode: No such file or directory
```

**Comment by kubukoz** [comment]
Is that some kind of recent change?

I still do this:

```nix
package = pkgs.runCommand "dummy" { } "mkdir $out" // {
  pname = pkgs.vscode.pname;
  version = "0.0.0";
};
```

Can you share the non-confidential parts of `/nix/store/zhrjg6wxrxmdlpn6iapzpp2z2vylpvw5-home-manager.sh` (including line 125 as per the error)

**Comment by OliverJAsh** [comment]
Ah, thanks, I needed this: `version = "0.0.0"`.

**Comment by nemith** [comment]
My company has MDM that insists on managing my VSCode.  package = null (along with specifying the flavor manually) would be perfect.  

Any reconsiderations here?
